File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.py
from itertools import combinations

class Solution:
    def answerQueries(self, nums: List[int], queries: List[int]) -> List[int]:
        nums.sort()
        answer = []
        for q in queries:
            count = 0
            for num in nums:
                if q >= num:
                    q -= num
                    count += 1
                    
            answer.append(count)
        return answer
                

File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.rs
impl Solution {
    pub fn answer_queries(mut nums: Vec<i32>, queries: Vec<i32>) -> Vec<i32> {
        nums.sort();
        let mut answer: Vec<i32> = vec![];
        
        for q in &queries {
            let mut count = 0;
            let mut total = *q;
            
            for num in &nums {
                if total >= *num {
                    total -= *num;
                    count += 1;
                }
            }
            answer.push(count);
        }
        return answer;
    }
}